### DISCUSSION TITLE
is the Instagram integration currently functional?

### DISCUSSION THREAD

**Discussion by mikespax** [chatwoot/chatwoot#8605] [Self hosted] [answered]
I hired someone on fiver to install this on my own server and set it up with Facebook and Instagram DMs. They're now telling me that the integration is broken and that there's nothing that can be done to get it working. Is this true, or should I find someone else to work on this for me?

I would really like this to work out, so please help. Thanks!

![Screenshots_2023-12-24-08-42-04](https://github.com/chatwoot/chatwoot/assets/14994710/936a6b6c-3775-4126-918d-47bc64648c50)

**Comment by Voldemorten** [comment]
I'm having an issue where, if you follow the chatwoot guide, it says to install a business app, however, Instagram Basic Display API is not available for business: [https://developers.facebook.com/docs/development/create-an-app/app-dashboard/app-types/](https://developers.facebook.com/docs/development/create-an-app/app-dashboard/app-types/). So, maybe. Maybe the integration is broken.

**Reply by UniHowCoder** [reply, marked as answer]
Basic Display API is being phased out : 
https://developers.facebook.com/blog/post/2024/09/04/update-on-instagram-basic-display-api/

Were you able to resolve the issue and connect instagram as self hosted?

**Reply by mikespax** [reply]
no, I gave up

**Comment by CrimsonSunrise** [comment]
I suppose Chatwoot needs to change the connection methods for the new "Instagram API with Instagram Login" or "Instagram API with Facebook login" to counter this problem.

**Reply by muhsin-k** [reply]
We have released a dedicated instagram channel using instagram business login in v4.1.0. ref:https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/instagram-via-instagram-business-login

**Comment by pranavrajs** [comment]
People who already have a working integration should not have any issues until December 4, 2024. However, new installations might not have access to the necessary permissions. 

The change regarding Instagram is new, please see the update from Meta [here](https://developers.facebook.com/blog/post/2024/09/04/update-on-instagram-basic-display-api/). Our team is currently assessing the required updates. This requires a significant change on our end, and we'll be shipping updates before the current permission is completely phased out.

**Reply by sequantum** [reply]
Hi. Is there any progress or an ETA for the updates? This integration is super important for my team's workflow. Thank you kindly.

**Comment by brundles** [comment]
@pranavrajs - so just for me to understand, is it currently possible to set up an instagram channel for a new installation?

(I've read the update from Meta, incl. the two API's with FB login/Insta login)

**Comment by sojan-official** [comment]
## Instagram API Update

The recent changes to the [Instagram Basic Display API](https://developers.facebook.com/blog/post/2024/09/04/update-on-instagram-basic-display-api/) (announced on September 4, 2024) should not impact Chatwoot, as we are not using the Basic Display API. We rely on the [Instagram Graph API](https://developers.facebook.com/blog/post/2024/07/30/instagram-api-with-instagram-login/), which has not been deprecated or affected by these changes.

Additionally, Instagram has simplified the process of connecting to their API via Instagram Login, eliminating the need to link to a Facebook page. This change, announced on July 30, 2024, allows Instagram to operate independently of Facebook, streamlining the integration flow. For more details, refer to the [Instagram API with Instagram Login update](https://developers.facebook.com/blog/post/2024/07/30/instagram-api-with-instagram-login/).

At this point, Chatwoot’s current integration with Instagram through the Graph API will continue to function as expected. We will be working on updating the login process with Instagram separately in the future.

**Comment by Funkelfetisch** [comment]
would anybody be willing to set this up for me?
I've wasted at least 3 man days on this, I give up.

Willing to pay up to 100€ /h

**Reply by CrimsonSunrise** [reply]
I Think it took around 5 or 6 weeks till I get mine to work. It is a tedious process sometimes because the permissions can take a long time to be given.
Just make sure you follow the steps described at [Setting up instagram account](https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/instagram-channel-setup/) and you should be fine.

**Reply by Funkelfetisch** [reply]
but how, if the last half of that manual involves a Meta product that no longer exists??

**Reply by CrimsonSunrise** [reply]
You would select "Instagram" instead of the non-existent "Instagram Graphic API". The following settings are very similar.

Also, you can ignore the part when it says you need to create a copy of your Facebook App. Just set up your app, request the permissions and you should be good to do.

**Reply by amandeepsinghkhalsa** [reply]
@Funkelfetisch Did you get permissions?
